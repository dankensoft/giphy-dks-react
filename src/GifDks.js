import React, {useState} from 'react';
import {CategoryAdd} from "./components/CategoryAdd";
import {GifGrid} from "./components/GifGrid";

export const GifDks = ({ defaultCategories = [] }) => {
    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategories);
    // const handleAdd = () => {
        // setCategories([...categories, 'HunterXHunter']);
        // setCategories(cats => [...cats, 'HunterXHunter']);
    // }
    return(
        <>
            <h2>GifDKS</h2>
            <CategoryAdd setCategories={setCategories}/>
            <hr />
            {/*<button onClick={handleAdd}>Add(+)</button>*/}
            <ol>
                {
                    categories.map(category => (
                        // return <li key={category}> {category} </li>
                        <GifGrid key={category} category={category}/>
                    ))
                }
            </ol>
        </>
    )
}
