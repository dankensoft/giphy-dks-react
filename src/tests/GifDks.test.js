import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from "enzyme";
import {GifDks} from '../GifDks'

describe('Pruebas en <GifDks />', () => {

  test('debe de mostrarse correctamente', () => {
    const wrapper = shallow(<GifDks />);
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['One Punch', 'Goku'];
    const wrapper = shallow(<GifDks defaultCategories = {categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })

})
