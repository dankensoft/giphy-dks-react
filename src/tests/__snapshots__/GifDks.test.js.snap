// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifDks /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifDKS
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifDks /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifDKS
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
